### DISCUSSION TITLE
Embed Monaco Editor in another editor or view

### DISCUSSION THREAD

**Discussion by kuldeeparora89** [eclipse-theia/theia#9360] [Q&A] [answered]
I am creating a Theia extension, in this I have a view where I need to include Monaco editor as input in the form. Purpose is to provide end-user a complete editing experience backed by LSP instead of plain input textbox.

What I have done till now -
vscode-extension for language support is created , language client & server is getting initiated by this .
My Language support is working on standalone file editor.

What is required now is to be able to open the Monaco editor on the fly & using it as a input in the form. Some of the use cases where we need to provide monaco editor as input - label editing in diagram editor , input in form based editor , input in property view etc.

What could be the best approach of achieving this ?

I have tried embedding it in my GLSP based editor on diagram nodes labels using below method - 
[https://github.com/eclipse-theia/theia/blob/master/packages/monaco/src/browser/monaco-editor-provider.ts#L519](url)

Monaco editor gets embedded & also opening up on label editing , but editor features like content assist , error highlighting etc. are not working .

Code -

```
export class MonacoEditorWidget {
@inject(MonacoEditorProvider)
protected readonly editorProvider: MonacoEditorProvider;

//embedding code
protected createInputElement() {
    const uri = new URI("inmemory:/model.xtree");
    const options={
        autoSizing: true,
        minHeight: 1,
        maxHeight: 10
    };
    this.editorProvider.createInline(uri,this.containerElement,options);
}

```

**Comment by vince-fugnitto** [comment, marked as answer]
I believe the question is a duplicate and has already been asked on discourse:
-  https://community.theia-ide.org/t/how-to-include-monaco-editor-in-div/1615/3.
